packages/shared/src/features/onboarding/types/funnel.ts

```diff
@@ -32,7 +34,13 @@ export const COMPLETED_STEP_ID = 'FINISH' as const;
 export type FunnelStepTransitionCallback<Details = Record<string, unknown>> =
   (transition: { type: FunnelStepTransitionType; details?: Details }) => void;
 
-type FunnelStepParameters = Record<string, string>;
+type FunnelStepParameters<
+  Params extends {
+    [p: string]: unknown;
+  } = Record<string, unknown>,
+> = {
+  [key in keyof Params]: Params[key];
+} & { [p: string]: unknown };
```

Comment:
Now we can extend the `parameters` prop in step on step props definition. Check `FunnelStepQuiz` for a sample.